x/ecocredit/server/testsuite/suite.go

```diff
@@ -83,6 +83,168 @@ func (s *IntegrationTestSuite) fundAccount(addr sdk.AccAddress, amounts sdk.Coin
+		},
+		{
+			name:   "valid",
+			msg:    ecocredit.MsgUpdateClassAdmin{ClassId: classID, Admin: admin.String(), NewAdmin: newAdmin},
+			expErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			updateRes, err := s.msgClient.UpdateClassAdmin(s.ctx, &tc.msg)
+			if tc.expErr {
+				s.Require().Error(err)
+				return
+			}
+
+			s.Require().NoError(err)
+			s.Require().NotNil(updateRes)
+
+			res, err := s.queryClient.ClassInfo(s.ctx, &ecocredit.QueryClassInfoRequest{ClassId: classID})
+			s.Require().NoError(err)
+			s.Require().NotNil(res)
+
+			s.Require().Equal(res.Info.Admin, newAdmin)
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestUpdateClassIssuers() {
+	admin := s.signers[0]
+	issuer1 := s.signers[1].String()
+	issuer2 := s.signers[2].String()
+	issuer3 := s.signers[3].String()
+
+	s.Require().NoError(s.fundAccount(admin, sdk.NewCoins(sdk.NewInt64Coin("stake", 4*ecocredit.DefaultCreditClassFeeTokens.Int64()))))
+	createClsRes, err := s.msgClient.CreateClass(s.ctx, &ecocredit.MsgCreateClass{Admin: admin.String(), Issuers: []string{issuer1}, Metadata: nil, CreditTypeName: "carbon"})
+	s.Require().NoError(err)
+	s.Require().NotNil(createClsRes)
+	classID := createClsRes.ClassId
+
+	testCases := []struct {
+		name   string
+		msg    ecocredit.MsgUpdateClassIssuers
+		expErr bool
+	}{
+		{
+			name:   "invalid: not admin",
+			msg:    ecocredit.MsgUpdateClassIssuers{ClassId: classID, Admin: issuer1, Issuers: []string{issuer1}},
+			expErr: true,
+		},
+		{
+			name:   "invalid: bad classID",
+			msg:    ecocredit.MsgUpdateClassIssuers{ClassId: "foobarbaz", Admin: admin.String(), Issuers: []string{}},
+			expErr: true,
+		},
+		{
+			name:   "valid",
+			msg:    ecocredit.MsgUpdateClassIssuers{ClassId: classID, Admin: admin.String(), Issuers: []string{issuer2, issuer3}},
+			expErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			updateRes, err := s.msgClient.UpdateClassIssuers(s.ctx, &tc.msg)
+			if tc.expErr {
+				s.Require().Error(err)
+				return
+			}
+
+			s.Require().NoError(err)
+			s.Require().NotNil(updateRes)
+
+			res, err := s.queryClient.ClassInfo(s.ctx, &ecocredit.QueryClassInfoRequest{ClassId: classID})
+			s.Require().NoError(err)
+			s.Require().NotNil(res)
+
+			s.Require().Equal(res.Info.Issuers, tc.msg.Issuers)
+
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestUpdateClassMetadata() {
+	admin := s.signers[0]
+	issuer1 := s.signers[3].String()
+
+	s.Require().NoError(s.fundAccount(admin, sdk.NewCoins(sdk.NewInt64Coin("stake", 4*ecocredit.DefaultCreditClassFeeTokens.Int64()))))
+	createClsRes, err := s.msgClient.CreateClass(s.ctx, &ecocredit.MsgCreateClass{Admin: admin.String(), Issuers: []string{issuer1}, Metadata: nil, CreditTypeName: "carbon"})
+	s.Require().NoError(err)
+	s.Require().NotNil(createClsRes)
+	classID := createClsRes.ClassId
+
+	testCases := []struct {
+		name   string
+		msg    ecocredit.MsgUpdateClassMetadata
+		expErr bool
+	}{
```

Comment:
We should add a test case for metadata exceeding max length.

Reply:
added a test for this in `msg_test.go`